### PR TITLE
Add unit tests for OAuth2Service classes

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
Added unit tests for OAuth2Service class.
There were test failures in the previously written test cases, so they were modified.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2TokenValidationServiceTest.java [hunk 1]
```diff
@@ -70,13 +78,14 @@
...
 
         tokenValidationService = new OAuth2TokenValidationService();
-
         mockStatic(TokenValidationHandler.class);
         when(TokenValidationHandler.getInstance()).thenReturn(mockedValidationHandler);
 
         mockStatic(OAuthComponentServiceHolder.class);
         when(OAuthComponentServiceHolder.getInstance()).thenReturn(mockedOAuthComponentServiceHolder);
         when(mockedOAuthComponentServiceHolder.getOAuthEventInterceptorProxy()).thenReturn(mockedOAuthEventInterceptor);
+        //mockStatic(OAuth2Util.class);
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2TokenValidationServiceTest.java:87]: remove this if not required

File: components/org.wso2.carbon.identity.oauth/src/main/resources/repository/conf/carbon.xml [hunk 2]
```diff
@@ -0,0 +1,684 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<!--
+ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.oauth/src/main/resources/repository/conf/carbon.xml:3]: let's have this year in the copyrights

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java [hunk 3]
```diff
@@ -176,26 +188,78 @@ public void testAuthorizeWithException() throws IdentityOAuth2Exception {
...
+                {"dummyAppState"}
+        };
+    }
+
+    @Test(dataProvider = "InvalidAppStatDataProvider")
+    public void testValidateClientInfoWithInvalidAppState(String appState) throws Exception {
+
+        OAuthAppDO oAuthAppDO = getOAuthAppDO("dummyGrantType", "dummyCallbackUrl",
+                "dummyTenantDomain", "dummyCallbackURI");
+        oAuthAppDO.setState(appState);
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java:250]: where do we use this `oAuthAppDO` object?

Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java:250]: The logic is written in getOAuthAppDO private method.
-> when(oAuthAppDAO.getAppInformation(clientId)).thenReturn(oAuthAppDO);

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java:2]: Let's not change the year unless we introduce this entire test class newly?

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java [hunk 5]
```diff
@@ -176,29 +189,83 @@ public void testAuthorizeWithException() throws IdentityOAuth2Exception {
     @Test(dataProvider = "ValidateClientInfoDataProvider")
     public void testValidateClientInfo(String grantType, String callbackUrl, String tenantDomain, String callbackURI)
             throws Exception {
-        when(oAuthServerConfiguration.getTimeStampSkewInSeconds()).thenReturn(3600L);
+
+        getOAuthAppDO(grantType, callbackUrl, tenantDomain, callbackURI);
+        OAuth2ClientValidationResponseDTO oAuth2ClientValidationResponseDTO = oAuth2Service.
+                validateClientInfo(clientId, callbackURI);
+        assertNotNull(oAuth2ClientValidationResponseDTO);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java:196]: Shall we assert the values returned by the DTO

File: components/org.wso2.carbon.identity.oauth/src/main/resources/repository/conf/carbon.xml (hunk 2 above)
Comment (emswbandara) [inline components/org.wso2.carbon.identity.oauth/src/main/resources/repository/conf/carbon.xml:3]: change to 2021

**Review by emswbandara — APPROVED** [review]

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java [hunk 6]
```diff
@@ -470,4 +631,88 @@ private void setUpRevokeToken() throws Exception {
...
+    public void testGetOauthApplicationStateWithIdentityOAuth2Exception() throws Exception {
+
+        mockStatic(IdentityTenantUtil.class);
+        when(IdentityTenantUtil.getTenantId(anyString())).thenReturn(1);
+        whenNew(OAuthAppDAO.class).withNoArguments().thenThrow(IdentityOAuth2Exception.class);
+        assertNull(oAuth2Service.getOauthApplicationState(clientId));
+    }
+
+    @Test
+    public void testGetOauthApplicationStateWithInvalidOAuthClientExceptionException() throws Exception {
```
Comment (ashendes) [inline components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java:658]: The word 'Exception' has been repeated in the method name. Please fix.

File: components/org.wso2.carbon.identity.oauth/src/main/resources/repository/conf/carbon.xml (hunk 2 above)
Comment (sachiniWettasinghe) [inline components/org.wso2.carbon.identity.oauth/src/main/resources/repository/conf/carbon.xml:3]: Change the year to 2021

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java (hunk 6 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ServiceTest.java:658]: Addressed with commit [9e575a3](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1691/commits/9e575a323c0604e9985d430a5b3fee4942d86cb1)

File: components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2TokenValidationServiceTest.java [hunk 7]
```diff
@@ -70,13 +78,13 @@
...
 
         tokenValidationService = new OAuth2TokenValidationService();
-
         mockStatic(TokenValidationHandler.class);
         when(TokenValidationHandler.getInstance()).thenReturn(mockedValidationHandler);
 
         mockStatic(OAuthComponentServiceHolder.class);
         when(OAuthComponentServiceHolder.getInstance()).thenReturn(mockedOAuthComponentServiceHolder);
         when(mockedOAuthComponentServiceHolder.getOAuthEventInterceptorProxy()).thenReturn(mockedOAuthEventInterceptor);
+
```
[the remark this answers is not in the document]
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2TokenValidationServiceTest.java:87]: Addressed with commit [9e575a3](https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1691/commits/9e575a323c0604e9985d430a5b3fee4942d86cb1)

File: components/org.wso2.carbon.identity.oauth/src/main/resources/repository/conf/carbon.xml (hunk 2 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.oauth/src/main/resources/repository/conf/carbon.xml:3]: Already resolved in master branch

**Pull request merged**